OneSignal/OneSignal-Android-SDK
# Replace FocusDelaySyncJobService by Worker

# Description
## One Line Summary
A JobService is being used for delaying the on_focus service call, JobServices are not stopped by the system, this makes the app process be alive in the background. Replace FocusDelaySyncJobService by a Worker

## Details

### Motivation
Avoid battery draining by the app in the background

# Testing
## Unit testing
Return tests to the previous state before this change https://github.com/OneSignal/OneSignal-Android-SDK/pull/1198

## Manual testing
Tested under Android 11.

1. Ensure that battery saving is off and the app is not restricted when in the background
2. launch the app and then minimize it to the background
3. Run adb shell dumpsys power
4. Check that non PARTIAL_WAKE_LOCK is alive for OneSignal

# Affected code checklist

# Checklist
## Overview
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR

## Testing

## Final pass
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/onesignal/onesignal-android-sdk/1515)
